File: src/main/java/us/flowdesigns/commands/Command_gcmd.java
package us.flowdesigns.commands;

import org.apache.commons.lang3.StringUtils;
import org.bukkit.ChatColor;
import org.bukkit.command.Command;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;

// Credit to TF

@CommandPermissions(source = SourceType.ONLY_IN_GAME)
@CommandParameters(description = "Makes a player execute a command", usage = "/<command>")
public class Command_gcmd extends BaseCommand
{
    @Override
    public boolean run(final CommandSender sender, final Player sender_p, final Command cmd, final String commandLabel, final String[] args, final boolean senderIsConsole)
    {
        if (!sender.hasPermission("nitrogen.gcmd"))
        {
            sender.sendMessage(Messages.MSG_NO_PERMS);
            return true;
        }

        if (args.length < 2)
        {
            sender.sendMessage("/gcmd <player> <command>");
            return true;
        }

        final Player player = getPlayer(args[0]);

        if (player == null)
        {
            sender.sendMessage(Messages.PLAYER_NOT_FOUND);
            return true;
        }

        final String outCommand = StringUtils.join(args, " ", 1, args.length);

        try
        {
            sender.sendMessage(ChatColor.GRAY + "Sending command as " + player.getName() + ": " + outCommand);
            if (server.dispatchCommand(player, outCommand))
            {
                sender.sendMessage(ChatColor.GRAY + "Command sent.");
            }
            else
            {
                sender.sendMessage(ChatColor.GRAY + "Unknown error sending command.");
            }
        }
        catch (Throwable ex)
        {
            sender.sendMessage(ChatColor.GRAY + "Error sending command: " + ex.getMessage());
        }

        return true;
    }
}

File: src/com/fusenetworks/fuse/commands/Command_gcmd.kt
package com.fusenetworks.fuse.commands

import org.apache.commons.lang3.StringUtils
import org.bukkit.Bukkit
import org.bukkit.ChatColor
import org.bukkit.command.Command
import org.bukkit.command.CommandExecutor
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player

// Credit to TF

abstract class Command_gcmd : CommandExecutor {
    fun run(sender: CommandSender, sender_p: Player, cmd: Command, commandLabel: String, args: Array<String>, senderIsConsole: Boolean): Boolean {
        if (!sender.hasPermission("fuse.gcmd")) {
            sender.sendMessage(Messages.MSG_NO_PERMS)
            return true
        }
        if (args.size < 2) {
            sender.sendMessage("/gcmd <player> <command>")
            return true
        }

        val player = Bukkit.getPlayer(args[0])

        if (player == null) {
            sender.sendMessage(Messages.PLAYER_NOT_FOUND)
            return true
        }

        val outCommand = StringUtils.join(args, " ", 1, args.size)

        try {
            sender.sendMessage(ChatColor.GRAY.toString() + "Sending command as " + player.name + ": " + outCommand)
            if (Bukkit.getServer().dispatchCommand(player, outCommand)) {
                sender.sendMessage(ChatColor.GRAY.toString() + "Command sent.")
            } else {
                sender.sendMessage(ChatColor.GRAY.toString() + "Unknown error sending command.")
            }
        } catch (ex: Throwable) {
            sender.sendMessage(ChatColor.GRAY.toString() + "Error sending command: " + ex.message)
        }

        return true
    }
}
